c7n/resources/glue.py

```diff
@@ -637,3 +638,41 @@ def get_resource_policy(self, r):
             policy = {}
         r[self.policy_annotation] = policy
         return policy
+
+
+@GlueDataCatalog.action_registry.register('remove-statements')
+class RemovePolicyStatement(RemovePolicyBase):
+    """Action to remove policy statements from Glue Data Catalog
+
+    :example:
+
+    .. code-block:: yaml
+
+           policies:
+              - name: remove-glue-catalog-cross-account
+                resource: aws.glue-catalog
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: remove-statements
+                    statement_ids: matched
+    """
+    permissions = ('glue:PutResourcePolicy',)
+    policy_annotation = "c7n:AccessPolicy"
+
+    def process(self, resources):
+        resource = resources[0]
+        client = local_session(self.manager.session_factory).client('glue')
+        p = resource.get(self.policy_annotation)
+        if p is None:
```

Comment:
if this requires a policy filter that should be in the validate method, not simply ignoring

Reply:
Thanks for pointing this out @kapilt . Thinking about it, I think it would be pretty safe to look for `cross-account` filter as part of validation.
Other option: Add policy in case its missing, but then we would end up having extra code on this action which would be unnecessary as `remove-statement` uses CrossAccountAccessFilter annotation key to process. Let me know your thoughts. For now I have added a validate function and a test as per your suggestion